src/util/system.cpp

```diff
@@ -432,20 +432,21 @@ const fs::path& ArgsManager::GetDataDir(bool net_specific) const
         path = GetDefaultDataDir();
     }
 
-    if (!fs::exists(path)) {
-        fs::create_directories(path / "wallets");
-    }
-
     if (net_specific && !BaseParams().DataDir().empty()) {
         path /= fs::PathFromString(BaseParams().DataDir());
-        if (!fs::exists(path)) {
-            fs::create_directories(path / "wallets");
-        }
     }
 
     return path;
 }
 
+void ArgsManager::EnsureDataDir(bool net_specific)
+{
+    fs::path path = GetDataDir(net_specific);
+    if (!fs::exists(path)) {
+        fs::create_directories(path / "wallets");
```

Comment:
1. Should we care about possible exceptions here?
2. What is the point to run this code when `net_specific == false`?

Reply:
> 2\. When run with `net_specific = false`, this will return the general `DataDir`, e.g. `$HOME/.bitcoin` and with `true` will return the network-specific DataDir e.g. `$HOME/.bitcoin/regtest`

> 2\. When run with `net_specific = false`, this will return the general `DataDir`, e.g. `$HOME/.bitcoin` and with `true` will return the network-specific DataDir e.g. `$HOME/.bitcoin/regtest`

Sorry, if my question was unclear. My point is whether we are interested in the `wallet` directory when `net_specific = false`?

Reply:
Now updated to `EnsureDataDirNet` which always creates the `wallets/` path if it's creating a new DataDir, but otherwise doesn't.